src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,360 @@
+              @editNameTranslation="editNameTranslation($event)"
+              @removeNameTranslation="removeNameTranslation($event)"
+              @nameUndo="undoNameTranslation($event)"
+            />
+          </v-flex>
+        </v-layout>
+        <v-layout pt-5>
+          <v-flex xs12>
+            <div class="action-btns">
+              <v-btn large color="primary"
+                id="name-translation-done"
+                :disabled="isAddingNameTranslation || !hasPendingChange"
+                @click="setNameTranslations()"
+              >
+                <span>Done</span>
+              </v-btn>
+              <v-btn large outlined color="primary"
+                id="name-translation-cancel"
+                @click="cancelNameTranslationCorrection()"
+                :disabled="isAddingNameTranslation"
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-flex>
+        </v-layout>
+      </v-flex>
+    </v-layout>
+  </div>
+</template>
+
+<script lang="ts">
+// Libraries
+import { Component, Vue, Prop, Watch, Emit, Mixins } from 'vue-property-decorator'
+
+// Components
+import { ConfirmDialog } from '@/components/dialogs'
+import { ListNameTranslation, AddNameTranslation } from '.'
+
+// Interfaces
+import { ActionBindingIF, ConfirmDialogType,
+  NameTranslationDraftIF, NameTranslationIF } from '@/interfaces'
+
+// Enums
+import { ActionTypes } from '@/enums'
+
+// Mixins
+import { CommonMixin } from '@/mixins'
+import { Action, Getter } from 'vuex-class'
+
+@Component({
+  components: {
+    AddNameTranslation,
+    ListNameTranslation,
+    ConfirmDialog
+  }
+})
+export default class NameTranslation extends Mixins(CommonMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationDialog: ConfirmDialogType
+  }
+
+  @Prop({ default: () => [] })
+  private nameTranslations!: NameTranslationDraftIF[]
+
+  private actionTypes = ActionTypes
+
+  // Properties
+  private draftTranslations: NameTranslationDraftIF[] = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasPendingChange (): boolean {
+    return this.draftTranslations.length !== this.nameTranslations.length ||
+      !this.draftTranslations.every((translation, index) => {
+        return this.nameTranslations[index].value === translation.value
+      })
+  }
+
+  private get hasNameTranslationChange (): boolean {
+    return this.draftTranslations.length > 0 &&
+      this.draftTranslations.filter(x => x.action).length > 0
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.draftTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.draftTranslations = this.nameTranslations
+      .filter(x => x.action !== 'added')
+      .map(a => {
```

Comment:
Nice use of 'map' :)

Reply:
So it looks like you don't undo added names? Is this Scott's design?

Reply:
Its not specified in design